src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs

```diff
@@ -247,7 +247,7 @@ private void AnalyzeFormatArgument(OperationAnalysisContext context, IOperation
 
             switch (argumentExpression)
             {
-                case ILiteralOperation { ConstantValue: { HasValue: false, Value: string constantValue } }:
+                case ILiteralOperation { ConstantValue: { HasValue: true, Value: string constantValue } }:
                     return constantValue;
                 case IInterpolatedStringOperation interpolated:
```

Comment:
I believe we should just check if `IInterpolatedStringOperation.ConstantValue` is null. We should not need to do any of this looping through the parts

Reply:
essentially, I don't understand why it can't just be this.

```csharp
case IInterpolatedStringOperation { ConstantValue: { HasValue: true, Value: string constantValue } }:
    return constantValue;
```

Reply:
(Also, as a note, I don't believe the `case INameOfOperation` when there's no constant value is possible to hit?)

Reply:
Oh, and another note: you almost certainly don't want to be checking for `IParenthesizedOperation` unless you have found a specific test case in VB where the parentheses matter.

Reply:
> essentially, I don't understand why it can't just be this.

replaced as such and seems like all tests pass this way. thanks!

Reply:
@333fred ok thanks, removed the two switch cases